concert/experiments/imaging.py

```diff
@@ -387,9 +380,102 @@ async def _set_sample_height(self, height):
     async def _set_vertical_shift_per_tomogram(self, shift):
         self._vertical_shift_per_tomogram = shift
 
+
+class SteppedSpiralTomography(SteppedTomography, Spiral):
+    def __init__(self, walker, flat_motor, tomography_motor, vertical_motor, radio_position, flat_position,
+                 camera, start_position_vertical, sample_height, vertical_shift_per_tomogram,
+                 num_flats=200, num_darks=200, num_projections=3000, angular_range=180 * q.deg, start_angle=0 * q.deg,
+                 separate_scans=True):
+        SteppedTomography.__init__(self, walker=walker, flat_motor=flat_motor, tomography_motor=tomography_motor,
+                                   radio_position=radio_position,
+                                   flat_position=flat_position,
+                                   camera=camera, num_flats=num_flats, num_darks=num_darks,
+                                   num_projections=num_projections, angular_range=angular_range,
+                                   start_angle=start_angle,
+                                   separate_scans=separate_scans)
+        Spiral.__init__(self, vertical_motor=vertical_motor, vertical_shift_per_tomogram=vertical_shift_per_tomogram,
+                        sample_height=sample_height, start_position_vertical=start_position_vertical)
+
+    async def _prepare_radios(self):
+        await asyncio.gather(
+            SteppedTomography._prepare_radios(self),
+            self._vertical_motor.set_position(await self.get_start_position_vertical())
+        )
+
+    async def _finish_radios(self):
+        if self._finished:
+            return
+        await self._prepare_darks()
+        await self._tomography_motor.stop()
+        await self._vertical_motor.stop()
+        await self._vertical_motor['motion_velocity'].restore()
```

Comment:
Couldn't all these be executed concurrently and gathered like the line below?

Reply:
The stops yes. Changing the Motion Velocity while the Motor is moving is forbidden or could result in strange behaviour.